### ISSUE TITLE
JSON Web Token (JWT) matches incorrectly

### ISSUE THREAD

**Issue by SkeletalDemise** [bee-san/pyWhat#211]
```
$ poetry run pywhat "C8.M."
Matched on: C8.M.
Name: JSON Web Token (JWT)
```
It's too broad and catches obvious non-JWTs. If it can't be less broad then make the rarity `0`.

**Comment by nodtem66** [comment]
I sugguest the new one: `(?i)^((?=eyJ)(?:[a-z0-9_=\\-]+){17,}\\.(?=e)(?:[a-z0-9_=\\-]+){2,}\\.(?:[a-z0-9_\\-\\+\\/=]{43,}))$` instead of the old RegExp was proposed by @amadejpapez 
https://github.com/bee-san/pyWhat/blob/69da611a69c3bbbd51cfe0078ca497e1bca7bd02/pywhat/Data/regex.json#L2379

## Reason
JWT consists of three parts (header, payload, signature) separated by `.`, e.g. `xxxxx.yyyyy.zzzzz`
I generated the bare minimum JWT:
```json
Header
{
  "alg": "HS256"
}
Payload
{}
```
Result: `eyJhbGciOiJIUzI1NiJ9.e30.ZRrHA1JJJW8opsbCGfG_HACGpVUMN_a9IV7pAx_Zmeo`
Then the minimum length of each part is **20, 3, 43** respectively. 

Since the encoding character used for `base64url` is [a-zA-z0-9_\-], then the RegExp of each part is
`(?:[a-z0-9_=\\-]{20,})` `(?:[a-z0-9_=\\-]{3,})` `(?:[a-z0-9_=\\-]{43,})`

Plus, I also suggest the stricter JWT that checks the first part of the header and payload.
Since **base64Url(`{x`)** and **base64Url(`{"x`)** definitely starts with `e` and `eyJ` respectively, `(?=eyJ)` and `(?=e)` was added to check the first part of the header and payload.

## TL;DR
new loose JWT: `(?i)^((?:[a-z0-9_=\\-]{20,})\\.(?:[a-z0-9_=\\-]{3,})\\.(?:[a-z0-9_=\\-]{43,}))$`
new strict JWT: `(?i)^((?=eyJ)(?:[a-z0-9_=\\-]{17,})\\.(?=e)(?:[a-z0-9_=\\-]{2,})\\.(?:[a-z0-9_=\\-]{43,}))$`
